File: benchmarks/testdata/python/worker.py
import struct
import sys

from inscriptis import get_text

stdin, stdout = sys.stdin.buffer, sys.stdout.buffer

while header := stdin.read(4):
    (n,) = struct.unpack(">I", header)
    out = get_text(stdin.read(n).decode()).encode()
    stdout.write(struct.pack(">I", len(out)) + out)
    stdout.flush()

File: benchmarks/testdata/node/worker.mjs
import { convert } from 'html-to-text';

let buf = Buffer.alloc(0);

process.stdin.on('data', (chunk) => {
  buf = Buffer.concat([buf, chunk]);

  while (buf.length >= 4 && buf.length >= 4 + buf.readUInt32BE(0)) {
    const n = buf.readUInt32BE(0);
    const out = Buffer.from(convert(buf.subarray(4, 4 + n).toString('utf8'), { wordwrap: 65 }));
    buf = buf.subarray(4 + n);

    const header = Buffer.alloc(4);
    header.writeUInt32BE(out.length);
    process.stdout.write(Buffer.concat([header, out]));
  }
});
